rero_ils/modules/selfcheck/api.py

```diff
@@ -220,23 +220,17 @@ def patron_information(barcode, **kwargs):
                 LoanState.ITEM_IN_TRANSIT_FOR_PICKUP,
                 LoanState.ITEM_ON_LOAN
             ]
-            loans = get_loans_by_patron_pid(patron.pid, filter_states)
-            for loan in loans:
+            sip2_summary_fields = current_app.config.get('SIP2_SUMMARY_FIELDS')
+            for loan in get_loans_by_patron_pid(patron.pid, filter_states):
                 item = Item.get_record_by_pid(loan.item_pid)
-                if loan['state'] == LoanState.ITEM_ON_LOAN:
-                    patron_account_information.get('charged_items', []).append(
-                        item.get('barcode'))
-                    if loan.is_loan_overdue():
-                        patron_account_information.get('overdue_items', []) \
+                field = sip2_summary_fields.get(loan['state'])
+                patron_account_information.get(field, []).append(
```

Comment:
You need to use setdefault. If the key does not exits it put the value in a new fresh created array which is not linked to the `patron_account_information`.